tests/utility/test_user_utils.py

```diff
@@ -0,0 +1,39 @@
+"""Test Case for User Management Utilities."""
```

Comment:
This seems to be redundant as you already test user_utils through user_api.

Reply:
No, I am mocking out the utility in the api tests.

Reply:
Don't mock utility from api test and mock only external sources(DB).

Reply:
Isnt unit testing about testing the unit? Thats why we have fixtures and mocks If there was no external source then we would only need to test the API and dont need to write test cases for anything else. That is not correct.

Reply:
Unit testing doesn't mean we have to have test files for individual classes. We need to test the module/component as a whole by mocking out the external boundaries. If we have unittest per class/per function, it would definitely hinder ability to refactor the code. Excessive mocking cause refactoring hard. There is a very good post on this. Please check: https://medium.com/@Cyrdup/unit-testing-youre-doing-it-wrong-407a07692989

Reply:
There are several opinions on unit testing. TDD states that every function needs to first have a test case before it is written. It also says that you should test your function without testing the dependencies. This will make sure that your function is doing what is intended.

Reply:
AFAIK, TDD focuses on test first approach(i.e. define tests to cover the spec) and write minimal functionality to pass the test and improve upon with further test cases. It never defines what is unit. Unit is subjective and doesn't mean always a class/function.

Reply:
When you do TDD, you focus on just the function that you are working on. It encourages you to not bother about other dependencies.  Unit testing like I said even earlier has not defined what a unit but there are multiple ways of doing it so my suggestion is go through the test cases that have been written. If you do not find value in them, it can be discussed. If you want to enforce something and it is being followed all across the project I am fine with that also.